### DISCUSSION TITLE
Allow users to use their own API

### DISCUSSION THREAD

**Discussion by CookSleep** [open-webui/open-webui#7293] [Ideas]
# Feature Request

**Is your feature request related to a problem? Please describe.**
I really like Open WebUI, but after all, deploying it requires certain conditions, and many friends are unwilling to bother with deployment or pay for a server capable of supporting Open WebUI.

I hope to directly share my deployed Open WebUI with them, but let them use their own API. However, this is completely impossible in the current version unless I give them administrator privileges.

**Describe the solution you'd like**
Allow users to use their own API. Perhaps an option can be added to the user's personal settings, and the API information they enter there can be added to the model list.

**Describe alternatives you've considered**
Grant users administrator privileges, but obviously, this is not ideal.

**Comment by CookSleep** [comment]
Supplement: The APIs and models added by users themselves should preferably be visible only to them

**Reply by ShallowDream724** [reply]
Support lao friend hhh
